src/physics/radiation/shortwave_radiation.jl

```diff
@@ -141,86 +151,83 @@ export OneBandShortwaveRadiativeTransfer
     "[OPTION] Ozone absorption in lower stratosphere (W/m^2)"
     ozone_absorp_lower::NF = 0
 end
+Adapt.@adapt_structure OneBandShortwaveRadiativeTransfer
 
 # generator function
 OneBandShortwaveRadiativeTransfer(SG::SpectralGrid; kwargs...) = OneBandShortwaveRadiativeTransfer{SG.NF}(; kwargs...)
 initialize!(::OneBandShortwaveRadiativeTransfer, ::PrimitiveEquation) = nothing
 
-# function barrier to unpack model
-function shortwave_radiative_transfer!(
-        column::ColumnVariables,
-        t,          # Transmissivity array
-        clouds,     # NamedTuple from clouds!
-        radiation::OneBandShortwaveRadiativeTransfer,
-        model::PrimitiveEquation
-    )
-    return shortwave_radiative_transfer!(column, t, clouds, radiation, model.planet)
-end
-
 """$(TYPEDSIGNATURES)
 One-band shortwave radiative transfer with cloud reflection and ozone absorption."""
-function shortwave_radiative_transfer!(
-        column::ColumnVariables,
+@propagate_inbounds function shortwave_radiative_transfer!(
+        ij,
+        diagn,
         t,          # Transmissivity array
         clouds,     # NamedTuple from clouds!
         radiation::OneBandShortwaveRadiativeTransfer,
-        planet::AbstractPlanet,
+        model,
     )
     (; ozone_absorp_upper, ozone_absorp_lower) = radiation
-    (; albedo_ocean, albedo_land) = column
     (; cloud_cover, cloud_top, stratocumulus_cover, cloud_albedo, stratocumulus_albedo) = clouds
-    (; cos_zenith, land_fraction, nlayers, flux_temp_downward, flux_temp_upward) = column
 
-    # Apply ozone absorption at TOA
-    D_TOA = planet.solar_constant * cos_zenith
-    D = max(zero(D_TOA), D_TOA - ozone_absorp_upper - ozone_absorp_lower)
-    flux_temp_downward[1] += D
-
-    # Clear sky portion until cloud top
-    @inbounds for k in 1:(cloud_top - 1)
-        D *= t[k]
-        flux_temp_downward[k + 1] += D
+    dTdt = diagn.tendencies.temp_tend_grid
+    p_s = diagn.grid.pres_grid_prev[ij]
+    nlayers = size(dTdt, 2)
+
+    @inbounds begin
```

Comment:
with the `@propagate_inbounds` macro we use now we automatically inline and inbound everything from the callside (where we have a `@inbounds`), so this is redundant here